src/zenml/integrations/slack/alerters/slack_alerter.py

```diff
@@ -253,64 +280,76 @@ def post(
             logger.error(f"SlackAlerter.post() failed: {response}")
             return False
 
     def ask(
-        self, message: str, params: Optional[BaseAlerterStepParameters] = None
+        self, question: str, params: Optional[BaseAlerterStepParameters] = None
     ) -> bool:
         """Post a message to a Slack channel and wait for approval.
 
         Args:
-            message: Initial message to be posted.
+            question: Initial message to be posted.
             params: Optional parameters.
 
         Returns:
             True if a user approved the operation, else False
         """
-        rtm = RTMClient(token=self.config.slack_token)
         slack_channel_id = self._get_channel_id(params=params)
 
-        approved = False  # will be modified by handle()
-
-        @RTMClient.run_on(event="hello")  # type: ignore
-        def post_initial_message(**payload: Any) -> None:
-            """Post an initial message in a channel and start listening.
+        client = WebClient(token=self.config.slack_token)
+        approve_options = self._get_approve_msg_options(params)
+        disapprove_options = self._get_disapprove_msg_options(params)
 
-            Args:
-                payload: payload of the received Slack event.
-            """
-            web_client = payload["web_client"]
-            blocks = self._create_blocks(message, params)
-            web_client.chat_postMessage(
-                channel=slack_channel_id, text=message, blocks=blocks
+        try:
+            # Send message to the Slack channel
+            response = client.chat_postMessage(
+                channel=slack_channel_id,
+                text=question,
+                blocks=self._create_blocks(question, params),
             )
 
-        @RTMClient.run_on(event="message")  # type: ignore
-        def handle(**payload: Any) -> None:
-            """Listen / handle messages posted in the channel.
-
-            Args:
-                payload: payload of the received Slack event.
-            """
-            event = payload["data"]
-            if event["channel"] == slack_channel_id:
-                # approve request (return True)
-                if event["text"] in self._get_approve_msg_options(params):
-                    print(f"User {event['user']} approved on slack.")
-                    nonlocal approved
-                    approved = True
-                    rtm.stop()  # type: ignore[no-untyped-call]
-
-                # disapprove request (return False)
-                elif event["text"] in self._get_disapprove_msg_options(params):
-                    print(f"User {event['user']} disapproved on slack.")
-                    rtm.stop()  # type: ignore[no-untyped-call]
-
-        # start another thread until `rtm.stop()` is called in handle()
-        rtm.start()
-
-        return approved
+            if not response["ok"]:
+                logger.error("Failed to send message to Slack channel.")
```

Comment:
Same as above, I think this should log the failure reason

Reply:
Fixed as well.